lib/components/app/responsive-webapp.js

```diff
@@ -268,10 +278,31 @@ const WebappWithRouter = withRouter(
  * so that Auth0 services are available everywhere.
  */
 class RouterWrapperWithAuth0 extends Component {
+  state = {
+    messages: null
+  }
+
+  async componentDidMount () {
+    this.updateMessages()
+  }
+
+  async componentDidUpdate (prevProps) {
+    this.updateMessages(prevProps)
+  }
+
+  async updateMessages (prevProps) {
+    // If the locale changes (or no prevProps are provided), update the messages for that locale.
+    const locale = this.props.locale
+    if (locale && (!prevProps || locale !== prevProps.locale)) {
+      this.setState({ messages: await loadLocaleData(locale, this.props.customMessages) })
+    }
+  }
```

Comment:
Only the `updateMessages` method really needs to be async (the others don't use the `await` keyword). Here's how I would change this (along with some other slight tweaks):

```suggestion
  componentDidMount () {
    this.updateMessages()
  }

  componentDidUpdate (prevProps) {
    this.updateMessages(prevProps.locale)
  }

  async updateMessages (previousLocale) {
    // If the locale changes (or no prevProps are provided), update the messages
    // for that locale.
    const {customMessages, locale} = this.props
    if (locale && (!previousLocale || locale !== previousLocale)) {
      const messages = await loadLocaleData(locale, customMessages)
      this.setState({messages})
    }
  }
```

Reply:
Updated that code in this commit: https://github.com/opentripplanner/otp-react-redux/pull/375/commits/f5a74fabe0e292e3a3f6cb17278548d063945c14